src/youtube2zim/utils.py

```diff
@@ -31,12 +31,29 @@ def load_json(cache_dir, key):
     if not fname.exists():
         return None
     try:
-        with open(fname, "r") as fp:
+        with open(fname) as fp:
             return json.load(fp)
     except Exception:
         return None
 
 
+def load_mandatory_json(cache_dir, key):
+    """load mandatory JSON collection from path or raise an error"""
+    fname = cache_dir.joinpath(f"{key}.json")
+    if not fname.exists():
+        raise Exception(f"JSON file at {fname} not found")
+    with open(fname) as fp:
+        return json.load(fp)
+
+
 def has_argument(arg_name, all_args):
     """whether --arg_name is specified in all_args"""
     return list(filter(lambda x: x.startswith(f"--{arg_name}"), all_args))
+
+
+def render_template(env: jinja2.Environment, template_name: str, **kwargs):
+    """render a Jinja template and ensures that result is a string"""
+    html = env.get_template(template_name).render(kwargs)
+    if not isinstance(html, str):
```

Comment:
How can this happen?

Reply:
This is probably just typing issue in Jinja2, and I will be able to get rid of this when updating the library.

Reply:
Please mention it in the upgrade ticket so it's not forgotten